src/test/java/com/mongodb/hibernate/SessionTests.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SessionTests {
+
+    private SessionFactory sessionFactory;
+
+    @BeforeEach
+    void setUp() {
+        sessionFactory = new Configuration().buildSessionFactory();
+    }
+
+    @AfterEach
+    void tearDown() {
+        sessionFactory.close();
+    }
+
+    @Test
+    void testSessionFactoryOpenSession() {
+        assertNotNull(sessionFactory.openSession());
+    }
+
+    @Test
+    void testInSession() {
+        sessionFactory.inSession(session -> assertTrue(true));
+    }
+
+    @Test
+    void testSessionFactoryFromSession() {
+        var value = sessionFactory.fromSession(session -> 1);
+        assertEquals(1, value);
+    }
+
+    @Test
+    void testSessionDoWork() {
+        sessionFactory.inSession(session -> session.doWork(connection -> assertTrue(true)));
+    }
+
+    @Test
+    void testSessionDoReturningWork() {
+        sessionFactory.inSession(session -> {
+            var value = session.doReturningWork(connection -> 1);
+            assertEquals(1, value);
+        });
+    }
```

Comment:
The higher level goal is to unblock `Session` after implementing `connection`. Behind the scenes, all the other session testing cases without explicit usage of `Connection` would fail without this PR's code changes.

If we only focus on the scope of this sub-task, I agree, it seems unnecessary. But after this sub-task is done, the parent "unblocking session" ticket might still need some testing. This class's main goal is for thorough testing of "unblocking session".